### ISSUE TITLE
Surprising prelude error caused by literal format mistake

### ISSUE THREAD

**Issue by dylanmc** [fortlang/fort#1]
When compiling this code:

```
qualifier Prelude = "lib/prelude.fort"

main = do
  taps = if
    argc > 1 -> atoi @(argv # 1)
    otherwise -> cast 0x0B0E `Int`

  print "taps = "
  println taps
```

I got this error:

```
[error]: expected type expression as second argument
     ╭──▶ lib/prelude.fort@50:21-50:24
     │
  50 │ curry = \f x y -> f (x, y)
     •                     ┬── 
     •                     ╰╸ (x, y)
     •
     │ Hint: (U 32, Sum {B0E: })
     │ Hint: eval steps:
     │       examples/curry-error.fort@3:8:  main = do
     │       examples/curry-error.fort@5:5:      argc > 1 -> atoi @(argv # 1)
     │       examples/curry-error.fort@6:18:      otherwise -> cast 0x0B0E `Int`
     │       lib/prelude.fort@50:19:  curry = \f x y -> f (x, y)
     │       
─────╯
```

Which now that I understand the error was the hex constant format, makes some sense (the first hint), but the overall error pointing at the prelude's curry totally threw me off.

So I'd suggest 1) allow uppercase hex constants, and 2) if possible, avoid printing errors in the prelude?

Back to implementing my first Fort program!

**Comment by fortlang** [comment]
Yes, indeed hex/oct/bin constants are just lower-case at the moment.  Will put that on the todo list along with adding underscores to numeric constants, e.g. 0xFF_FF_FF_FF

Looking at the type error "stack trace" can be confusing, I agree.  On the other hand, I'd prefer not to make the prelude "special".  Not sure what the best thing here is.  Perhaps putting a line break in when the trace switches files would help?

**Comment by fortlang** [comment]
fixed in commit 54fd6798020cfe3d87f4c065857037d4afc7c217 (tag: v0.1.4.0-alpha)
